core/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -880,8 +880,15 @@ public FlushStats flushStats() {
     }
 
     public DocsStats docStats() {
```

Comment:
I think this should rather be something like this:

```Java

public DocsStats docStats() {
        long numDocs = 0;
        long numDeletedDocs = 0;
        long sizeInByte = 0;
        List<Segment> segments = segments(false);
        for (Segment segment : segments) {
            if (segment.search) {
                numDocs += segment.getNumDocs();
                numDeletedDocs += segment.getDeletedDocs();
                sizeInByte += segment.getSizeInBytes();
            }
        }
        return new DocsStats(numDocs, numDeletedDocs, sizeInByte);
    }
```
that way we maintain a consistent total and we can calculate the average at read time and aggregation of doc stats will be much simpler? I also think we should make sure the size in bytes is based on the currently used reader which is guaranteed by the `Segment#search` flag. WDYT?

Reply:
Yes, this makes the DocsStats simpler and the average value more accurate. I have updated this in https://github.com/elastic/elasticsearch/pull/27117/commits/662f062973bca3877b8c78b5acb6c75a6632fe8e